src/makeCalculator.js

```diff
@@ -4,7 +4,41 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(callback, num) {
+      callback.call(this, num);
+
+      return this;
+    },
+
+    add(num) {
+      this.result += num;
+    },
+
+    subtract(num) {
+      this.result -= num;
+    },
+
+    multiply(num) {
+      this.result *= num;
+    },
+
+    divide(num) {
+      if (num === 0) {
+        return "Can't divide by zero";
+      }
+
+      this.result /= num;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
```

Comment:
The `reset` method correctly resets the `result` to 0 and allows chaining by returning `this`.